lib/peep.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'database_connection'
+
+class Peep
+  attr_reader :id, :content, :posted, :username, :name
+
+  def initialize(id:, content:, username:, name:, posted: Time.now.strftime("%I:%M %p, %d/%m/%Y"))
+    @id = id
+    @content = content
+    @posted = posted
+    @username = username
+    @name = name
+  end
+
+  def self.all
+    result = DatabaseConnection.query("SELECT * FROM peeps;")
+    array_of_peeps = result.map do |peep| 
+      Peep.new(id: peep['id'], content: peep['content'], 
+               posted: peep['posted'], username: peep['username'], name: peep['name'])
+    end
+    array_of_peeps.reverse
+  end
+
+  def self.create(content:, username:, name:)
+    posted = Time.now.strftime("%I:%M %p, %d/%m/%Y")
```

Comment:
Just a thought - where is it best to convert a timestamp to a string? You have done it when the peep is created so that what is in the database is a string. Perhaps it would maintain more information to keep the timestamp persisting in the database and the model, and only convert it to the string you want when taking it to the view (not sure if that would be in the controller or the view you would do that)...